src/Networking/v2/Models/Subnet.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace OpenStack\Networking\v2\Models;
+
+use OpenStack\Common\Resource\AbstractResource;
+use OpenStack\Common\Resource\Listable;
+use OpenStack\Common\Resource\Creatable;
+use OpenStack\Common\Resource\Deletable;
+use OpenStack\Common\Resource\Retrievable;
+
+/**
+ * Represents a Networking v2 Network.
+ *
+ * @property \OpenStack\Networking\v2\Api $api
+ */
+class Subnet extends AbstractResource implements Listable, Retrievable, Creatable, Deletable
+{
+    public $id;
+    public $name;
+    public $enableDhcp;
+    public $networkId;
+    public $dnsNameservers;
+    public $allocationPools;
+    public $hostRoutes;
+    public $ipVersion;
+    public $gatewayIp;
+    public $cidr;
+    public $tenantId;
+    public $links;
+
+    protected $aliases = [
+        'enable_dhcp' => 'enableDhcp',
+        'network_id' => 'networkId',
+        'dns_nameservers' => 'dnsNameservers',
+        'allocation_pools' => 'allocationPools',
+        'host_routes' => 'hostRoutes',
+        'ip_version' => 'ipVersion',
+        'gateway_ip' => 'gatewayIp',
+        'tenant_id' => 'tenantId'
+    ];
+
+    protected $resourceKey = 'subnet';
+    protected $resourcesKey = 'subnets';
+
+    /**
+     * {@inheritDoc}
+     */
+    public function retrieve()
+    {
+        $response = $this->execute($this->api->getSubnet(), ['id' => (string) $this->id]);
+        $this->populateFromResponse($response);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param array $data {@see \OpenStack\Networking\v2\Api::postSubnets}
+     */
+    public function bulkCreate(array $data)
+    {
+        $response = $this->execute($this->api->postSubnets(), [
+            'subnets' => $data,
+        ]);
+        $body = $response->json();
+        $json = $body['subnets'];
+
+        $subnets = [];
+        foreach($json as $resourceData) {
+            $resource = $this->newInstance();
+            $resource->populateFromArray($resourceData);
+            $subnets[] = $resource;
+        }
+
+        return $subnets;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param array $data {@see \OpenStack\Networking\v2\Api::postSubnet}
+     */
+    public function create(array $data)
+    {
+        $response = $this->execute($this->api->postSubnet(), $data);
+        return $this->populateFromResponse($response);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function update()
+    {
+        $response = $this->executeWithState($this->api->putSubnet());
+        return $this->populateFromResponse($response);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function delete()
+    {
+        $this->execute($this->api->deleteSubnet(), $this->getAttrs(['id']));
```

Comment:
nothing wrong with this, but I always use `executeWithState` whenever possible, since all necessary attributes will be picked up according to the API definition, rather than us manually remembering. Not really a big issue here since it's obvious that `id` is the only attr